.pre-commit-config.yaml

```diff
@@ -0,0 +1,51 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.3.0
+  hooks:
+  - id: trailing-whitespace
+    exclude: '.*/Baseline'
+  - id: end-of-file-fixer
+    exclude: '.*/Baseline'
+  - id: check-yaml
+  - id: check-added-large-files
+
+- repo: https://github.com/psf/black.git
```

Comment:
Is this for the usual everybody-formats-similarly reasons, or is there a template CI angle to it? Have we agreed that black is the formatter of choice for Zeek project code? If so, we may want to introduce it via a dedicated set of PRs, since neither zkg nor zeek-script currently use it. Might be worth a Discussion ticket for reference, too.

Reply:
I introduced this to remove the formatting element from reviews completely (at least for Python files). I don't care much about the particular style, but went with Black since it is most aggressive in formatting; pure formatting changes to a Black-formatted file likely would result in the edits being removed again while e.g., yapf might leave them in.

AFAIK we don't really have a Zeek Python style and I went with yapf over in zeek/zeek to be least intrusive (given how loosly formatting was handled there historically). I would have expected that most people making edits here wouldn't be surprised by Black formatting stuff some way either, but just adopt its changes, so I felt that using a more aggressive formatter should be fine. I can go with e.g., yapf if you feel this goes too far.

In any case, we should have some formatter for Python sources.

Reply:
Actually, thinking back to it, I used yapf over in zeek/zeek since allowed adjusting the formatting style to minimize the diff over the existing style. I am still unsure whether that was _only to minimize the diff_ or whether there's some genuine Zeek Python formatting style we care about deeply.

Reply:
And just to clarify, this is for our own repo, we do not force this formatter on anything created from any templates in here.

Reply:
> In any case, we should have some formatter for Python sources.

Yeah I agree with this, don't get me wrong. I'm just not a fan of introducing the formatting as a requirement in an unrelated PR, and I wonder whether we should at least consider yapf everywhere since we're using it in Zeek. So can I suggest that we remove this here for now, start a discussion, and then apply what we land on across everything? I can even see using black for projects other than Zeek, but we should have that written down somewhere. 

>  we do not force this formatter on anything created from any templates in here.

Yep, thanks!

Reply:
I have taken out any formatting changes here 🤷